### PR TITLE
fix(components/Datalist): restricted mode

### PR DESCRIPTION
**Pull request by sgendre**
**What is the problem this PR is trying to solve?**
Restricted props prevents onBlur matching between inputted string and available items

**What is the chosen solution to this problem?**
Skip that unrelated condition.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/src/UIForm/fields/Datalist/__snapshots__/Datalist.component.test.js.snap [hunk 1]
```diff
@@ -27,7 +27,7 @@ exports[`Datalist component should render 1`] = `
     onFocus={[Function]}
     placeholder="Type here"
     readOnly={false}
-    restricted={false}
+    restricted={true}
```
Comment (ncomont) [inline packages/forms/src/UIForm/fields/Datalist/__snapshots__/Datalist.component.test.js.snap]: seems like the default value assigned in `get(this.props, 'schema.restricted', false)` is not taken into account, right ?

Comment (sgendre) [reply packages/forms/src/UIForm/fields/Datalist/__snapshots__/Datalist.component.test.js.snap]: I think you might look at the wrong place, both schema and schemaMultiSection do have `restricted: true`

Comment (ncomont) [reply packages/forms/src/UIForm/fields/Datalist/__snapshots__/Datalist.component.test.js.snap]: oh you're right, my bad !

File: packages/forms/src/UIForm/fields/Datalist/Datalist.component.js [hunk 2]
```diff
@@ -165,6 +165,7 @@ class Datalist extends Component {
 						'aria-required': this.props.schema.required,
 						'aria-describedby': `${descriptionId} ${errorId}`,
 					}}
+					restricted={get(this.props, 'schema.restricted', false)}
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/Datalist/Datalist.component.js]: this is supposed to be in options.

File: packages/components/src/Datalist/Datalist.component.js [hunk 3]
```diff
@@ -308,6 +308,13 @@ class Datalist extends Component {
 		});
 	}
 
+	clearValue() {
```
Comment (jsomsanith-tlnd) [inline packages/components/src/Datalist/Datalist.component.js]: You can clear that

**Pull request merged**